### DISCUSSION TITLE
DNS queries every minute despite period being set to 30m

### DISCUSSION THREAD

**Discussion by JonnyBooker** [qdm12/ddns-updater#303] [Q&A]
I've recently tried out DDNS updater with my Synology as a Docker container. Synology DDNS didn't support Namecheap so thought I'd try this!

It all seems to work and correctly does update my DNS record. I have set the config to update with a `PERIOD` of `30m`. However, I'm running PiHole on my network and can see, its running DNS queries every minute and this is adding up to be quite chatty on my network. Its not so much a problem but wondered as to why it is so chatty. Example in screenshot of Pi Hole to show its making DNS queries against PiHole (also running on Synology).

![image](https://user-images.githubusercontent.com/1131478/152023047-19e752b2-680d-46e8-9ae5-e0b873c86172.png)

Searching the logs of the container shows no active requests going off. And the config is definitely set to 30m:
![image](https://user-images.githubusercontent.com/1131478/152023705-0b8d6951-7be9-4ac2-83fe-b4e482e5ac26.png)

Wondering if any thoughts on this

Edit: In 24 hours, I've had 2000+ DNS queries

**Comment by maboxx** [comment]
Hello, I would like to join here. I also see in my pi-hole the chattiness of the DDNS Updater. In 24h this is about 3500 request.
Would be great if the issue could be discussed and maybe can be solved. This traffic would not have to be.

**Comment by qdm12** [comment]
It should resolve your domain on every period and compare it with your current public IP address.
Although that should only happen 48 times a day for a 30min period.

It also fetches your public IP address using DNS though (which you can disable by selecting only `PUBLICIP_FETCHERS=http` as the fetch method).

You could use `LOG_LEVEL=debug` to see a bit more what it's doing?

**Comment by qdm12** [comment]
Actually the health check does a resolution

https://github.com/qdm12/ddns-updater/blob/5b426afc57148c2fb7d8d9acd7338a9d977f5511/internal/health/check.go#L31

So that's most likely why it triggers so often. You could change the period of the healthcheck or feel free  propose to change the health check code!

**Reply by JonnyBooker** [reply]
Ahhh ok. What option would I be able to use to change the period of the healthcheck? Looking through the configuration/health check code, I can't see an option to configure this at all unless I have missed it. I don't have much experience with Go admittedly.

**Reply by maboxx** [reply]
I have the same issue, would appreciate a response.

**Reply by qdm12** [reply]
See https://docs.docker.com/engine/reference/run/#healthcheck

I guess `--health-interval 5m` would do

**Reply by JonnyBooker** [reply]
Thanks for the feedback and pointing me in the direction for what needed to change. For my use, I forked your repo and changed the `--health-interval` to be 60m for the time being for my use to reduce the network noise and am using this. 
https://github.com/JonnyBooker/ddns-updater

I think this might be for the more specific instance that I'm running PiHole and DDNS Updater on my Synology

**Reply by qdm12** [reply]
Yes that makes sense. Although you can configure this health period in your run command or docker-compose.yml file instead of rebuilding the image/modify the Dockerfile.

Although if I may add, you might want to be careful with pihole caching. What I do personally is I run the container with `--dns 1.1.1.1` since I don't care about privacy, that way it always works too, even if my pihole/other dns fails, and also it doesn't mess up statistics of my dns.
